rusk/src/lib/chain/rusk.rs

```diff
@@ -117,6 +125,21 @@ impl Rusk {
                 continue;
             }
 
+            // check free transaction
+            if unspent_tx.inner.nullifiers.is_empty() {
```

Comment:
This checks must be added to the accept_transactions and verify_transactions too.
Otherwise only the block generator is currently checking for free transactions.

I strongly suggest to add a test where an invalid block is produced (with invalid free transactions) which is discarded during the acceptance phase